product_customer_code_sale/models/sale.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Nicola Malcontenti - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from openerp import models, api, fields
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    @api.depends('product_id', 'order_id.partner_id')
```

Comment:
Suggestion.
Why not use @ api.one and it would look like this?
@api.one
@api.depends('product_id', 'order_id.partner_id')
def _compute_product_customer_code(self):
product_customer_code_obj = self.env['product.customer.code']
partner = self.order_id.partner_id
product = self.product_id
code = ''
if product and partner:
code_ids = product_customer_code_obj.search([
('product_id', '=', product.id),
('partner_id', '=', partner.id),
], limit=1)
if code_ids:
code = code_ids[0].product_code or ''
self.product_customer_code = code